### PR TITLE
feat(KalturaAPI): added ibm namespace to event-listeners

### PR DESCRIPTION
**Pull request by tweenn**
### Related Ticket(s)

[#7488 - Kaltura Player Metrics: For CTA clicks (& other events) - DDS Consulting](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/7488)
[#7652 - feat(KalturaAPI): services listeners-exposure and extra metrics](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/pull/7652)

### Description

I'm opening this PR as a follow-up from [#7652](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/pull/7652) that had an issue with extra metrics for quartiles and seek (drag and drop the scrubber).

Said that, I want to point out that there shouldn't be a breaking change in this PR - except by the error addressed in 388d15e3b4047a8a9d2d4df3b5a844c1c2f23a05 (more on changelogs)

And this PR addresses 3 (major but not breaking) changes in the kaltura-service - and analytics-video portion:

1. Make IBM Metrics firing optional - but turned on by default
2. Expose the readyCallback for custom manipulations of the player and their event-listeners
3. Add extra event listener for Mediacenter's cta-clicks and cta-impressions

It also has some unit-testing and coverage bumping, a bug fix in the analytics file (found while writing tests).

And I tried my best to make each commit be as semantic as possible with the code implemented - meaning the changelog may feel redundant.
 
### Changelog

* 52dc06812bb236bef1e45b6ebd7d4c257b44d78d
Added namespaces to the player event-listeners.
This will keep IBM listeners in a safe position if a developer try to remove/unbind listeners, as like using a `kdp.removeListener('seek')` will remove all global/generic `'seek'` listeners, but will not remove the `'seek.ibm'` listener.

* 3c7c1835be59637153e73804e68683ebadb81d06
Added a param to `embedMedia` to make the IBM Metrics listeners optional - turned on by default
And fixed a documentation error where flashvars was set to boolean, when it was actually an object

* 1ddf75a61a2e884cd3a369143d65f17bea74d721
Added a param to `embedMedia` that is a function that receives the `KDP` object in order to expose the `readyCallback`.

* 388d15e3b4047a8a9d2d4df3b5a844c1c2f23a05
The `AnalyticsAPI.videoPlayerStats` prop `executionPath` was expecting a `params.videoId`, that the old `VideoPlayer` service sent, but since the deprecation of that service in favor of `KalturaPlayer`, this param was changed/renamed to `mediaId` - and thus the metrics being fired right now are basically sending `{ executionPath: undefined }` to DBDM

* 69ab26a0c98ff7460f21adc6038ca7dd59db03e7
Added Mediacenter's CTA-CLICK listener and custom metric data to `KalturaPlayer` and the appropriate behaviour over the `AnalyticsAPI.videoPlayerStats` to add custom metrics if available.

* 74f2f2ce9d2c0620f9a5f47a3875d60063ece097 & 8c05d04434fda0fb14d716fd003f04d2662bbb10
Test and Coverage bumping - because I didn't like the values I saw 🤣 

* fb7109de74fbe0590141346a43d89621a0cbea26
Add retrocompatibility for the fix done in 388d15e3b4047a8a9d2d4df3b5a844c1c2f23a05 - which would insert a breaking change otherwise - plus I've added a test case in Analytics's test scenarios for this change AND cta-clicks

-----------

I have read the DCO document and I hereby sign the DCO.

### CODE REVIEW

**Comment by tweenn** [comment]
@jeffchew asked in the previous PR #7652 [here](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/pull/7652#pullrequestreview-813071455) the following:
> Not sure if we wanted to include as part of this PR or perhaps a separate PR, if I recall we were discussing also creating standard event listeners that any application could also subscribe to if they wanted to monitor these behaviors (e.g. append their own custom tracking triggers), which would be triggered by these Kaltura listeners. I'd be fine with adding this in a separate PR though.

The `readyCallback` method under the `kWidget.embed` will be executed as soon as the player iframe is created, which you can use to set listeners or other player-driven configuration.
So, instead of adding some custom events, I thought on executing a function inside the readyCallback that passes the `kdp` object would allow customization of such execution context without limiting the consumer to some specific events - and that's what commit [#1ddf75a61a2e884cd3a369143d65f17bea74d721](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/commit/1ddf75a61a2e884cd3a369143d65f17bea74d721) is about.

Since the global kWidget object allow the removal of any listener associated to a player, kaltura defines as a best practice the addition of a namespace to each listener so a global removal wouldn't affect our code - which is the change made in [#52dc06812bb236bef1e45b6ebd7d4c257b44d78d](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/pull/7789/commits/52dc06812bb236bef1e45b6ebd7d4c257b44d78d).

And finally, for those consumers who don't want IBM analytics over a Gartner (or other partners) video, I've added a bool (that by default if true) to define if the namespaced events we have for metrics should be added or not to the player.
PS: Gartner videos that IBM buy the rights to stream are kaltura based

In short, we have some scenarios covered like:
* "I don't want IBM's analytics listeners that are on by default, here's my own events"
* "Here's my own events, which will remove one specific listener from IBM and nothing more"

And by the way - this changes allow a great level of refactoring to the audio player component - that sets a custom readyCallback through the global `kWidget.addReadyCallback()` method which is not very memory-friendly

File: packages/services/src/services/KalturaPlayer/KalturaPlayer.js [hunk 1]
```diff
@@ -234,17 +243,33 @@ class KalturaPlayerAPI {
...
+                fireEvent({
+                  playerState: 101,
+                  kdp,
+                  mediaId,
+                  customMetricsData,
+                });
+              });
+            }
+
+            customReadyCallback(kdp);
```
Comment (jeffchew) [inline packages/services/src/services/KalturaPlayer/KalturaPlayer.js:272]: 👍

File: packages/services/src/services/KalturaPlayer/KalturaPlayer.js [hunk 2]
```diff
@@ -177,7 +177,10 @@ class KalturaPlayerAPI {
    *
    * @param {string} mediaId  The mediaId we're embedding the placeholder for.
    * @param {string} targetId The targetId the ID where we're putting the placeholder.
-   * @param {boolean} flashvars Determine any extra param or plugin for the player.
+   * @param {object} flashvars Determine any extra param or plugin for the player.
+   * @param {boolean} useIbmMetrics Whether or not should IBM Metrics events be fired.
+   * @param {Function} customReadyCallback Determine any extra functions that should be executed
```
Comment (jeffchew) [inline packages/services/src/services/KalturaPlayer/KalturaPlayer.js:182]: Minor thing, but can add a note that the player object is passed in the custom method in order to add additional custom functionality (e.g. listeners, etc). Is there a link we can send users to for a full API that they can tap into?

File: packages/services/src/services/Analytics/Analytics.js [hunk 3]
```diff
@@ -267,14 +267,26 @@ class AnalyticsAPI {
...
+      eventData.playerStateLabel = data.customMetricsData.playerStateLabel;
+    }
+
+    if (data?.customMetricsData?.driverId) {
+      eventData.driverId = data.customMetricsData.driverId;
+    }
+
+    if (data?.customMetricsData?.targetURL) {
+      eventData.targetURL = data.customMetricsData.targetURL;
+    }
```
Comment (emyarod) [inline packages/services/src/services/Analytics/Analytics.js:288]:
```suggestion
      driverId: data?.customMetricsData?.driverId,
      executionPath: data.videoId || data.mediaId,
      execPathReturnCode: playerState,
      eventVidStatus: data.playerState,
      eventVidTimeStamp: currentTime,
      eventVidLength: duration,
      eventVidPlayed: percentWatched + '%',
      playerStateLabel: data?.customMetricsData?.playerStateLabel,
      targetURL: data?.customMetricsData?.targetURL,
    };
```

would we be able to move the custom metrics data logic into the `eventData` declaration like this?

Comment (tweenn) [reply packages/services/src/services/Analytics/Analytics.js:288]: @emyarod - As far as this was implemented on v18 (in last year) and as far as I understood, the 3 params (playerStateLabel, driverId and targetURL) were to be used exclusively with this event firing, not on all of then.

And by declaring the property on object initiating means that it will be fired as `undefined`, while having it under the above if statement, the object will have the property added only if the param is passed.

**Review by jeffchew — APPROVED** [review]
The comments I had were pretty minor and think we can just merge as-is, let me know if there was anything else you wanted to have addressed @emyarod
